static/js/filter.js

```diff
@@ -1,7 +1,95 @@
 (function() {
 	"use strict";
-
     
-	
-	
-})();
\ No newline at end of file
+	const filterEmailAddresses = [];
+	const filterAliasLabels = [];
+	const aliasesWithLabelsCollection = [];
+    const aliasCollection = [];
+    const aliases = document.querySelectorAll(".c-alias");
+
+    const filterLabelTotalCases = document.querySelector(".js-filter-case-total");
+    const filterLabelVisibleCases = document.querySelector(".js-filter-case-visible");
+
+    const filterResetButton = document.querySelector(".js-filter-reset")
+
+    const filterInput = document.querySelector(".js-filter-search-input");
+
+	function filterInputWatcher(e) {
+		const query = e.target.value.toLowerCase();
+
+        // if (query === "") {
+        //     resetFilter();
+        //     return;
+        // }
+
+		// Hide all cases
+		aliases.forEach(alias => {
+            alias.style.display = "none";
+        });
+
+		// Loop through both arrays, making that alias visible if it is a match.
+        const matchListEmailAddresses = filterEmailAddresses.filter(s => s.includes(query));
+        const matchListAliasLabels = filterAliasLabels.filter(s => s.includes(query));
+
+        // Set the current number of "found" results
+        // TODO: Reset this number back to `aliases.length` when the search field is empty
+        if ( (matchListEmailAddresses.length + matchListAliasLabels.length) <= aliases.length ) {
+            filterLabelVisibleCases.textContent = matchListEmailAddresses.length + matchListAliasLabels.length;
+        }
+
+        for (const alias of matchListEmailAddresses) {
+            let index = filterEmailAddresses.indexOf(alias);
+            if (index >= 0) {
+                aliasCollection[index].style.display = "block";
+            }
+        }
+
+        // TODO: Map the entire list of aliases better so we dont have two seperate DOM collection arrays to loop through.
+        for (const alias of matchListAliasLabels) {
+            let index = filterAliasLabels.indexOf(alias);
+            if (index >= 0) {
+                aliasesWithLabelsCollection[index].style.display = "block";
+            }
+        }
```

Comment:
Could try to refactor these 2 functions into a single function that accepts `matchListEmailAddresses, aliasCollection` and then `matchListAliasLabels, aliasesWithLabelsCollection` as arguments?

But that might end up being more complex/confusing to read?

Reply:
My TODO really states it – we're actually looping through two different arrays. There's an entirely better way to do this. I would recommend leaving it as is for the moment. 

Note that I did find an edge case in the logic where if you name two aliases the same thing, it will only count/display the first one. That alone is worth fixing this up into a more flexible system. (Maybe a `Map`)?